test/integration/liquid/register-filters.spec.ts

```
@@ -25,14 +25,26 @@ describe('liquid#registerFilter()', function () {
       const src = `{{ userId | get_user_data | json }}`
-      const dst = '{"userId":"alice","userName":"ALICE"}'
-      const html = await liquid.parseAndRender(src, { userId: 'alice' })
-      return expect(html).toBe(dst)
+      await expect(liquid.parseAndRender(src, { userId: 'alice' }))
+        .rejects.toThrow(/async filter "get_user_data" is not supported in the sync-only build/)
+    })
+    // SYNC FORK: generator/async-function filters are rejected at registration time
+    it('should reject async function filter at registration time', () => {
+      // NOTE: eval is used because ts-jest compiles `async function` down to a
+      // plain function at the ES6 target; eval'd code keeps the native AsyncFunction.
+      const asyncFn = eval('(async function (v) { return v })')
```

Biome (2.5.6)

[error] eval() exposes to security risks and performance issues.

(lint/security/noGlobalEval)